plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/DeltaLakeParquetStatisticsUtils.java

```diff
@@ -222,6 +237,37 @@ private static Map<String, Object> jsonEncode(Map<String, Optional<Statistics<?>
                 .collect(toImmutableMap(Map.Entry::getKey, entry -> entry.getValue().get()));
     }
 
+    public static Map<String, Object> toNullCounts(Map<String, Type> columnTypeMapping, Map<String, Object> values)
+    {
+        ImmutableMap.Builder<String, Object> nullCounts = ImmutableMap.builderWithExpectedSize(values.size());
+        for (Map.Entry<String, Object> value : values.entrySet()) {
+            Type type = columnTypeMapping.get(value.getKey());
+            nullCounts.put(value.getKey(), toNullCount(type, value.getValue()));
+        }
+        return nullCounts.buildOrThrow();
+    }
+
+    private static Object toNullCount(Type type, Object value)
+    {
+        if (type instanceof RowType rowType) {
+            Block rowBlock = (Block) value;
+            ImmutableMap.Builder<String, Object> nullCounts = ImmutableMap.builderWithExpectedSize(rowBlock.getPositionCount());
+            for (int i = 0; i < rowBlock.getPositionCount(); i++) {
+                RowType.Field field = rowType.getFields().get(i);
+                Object nullCount;
+                if (field.getType() instanceof RowType) {
+                    nullCount = rowBlock.getChildren().get(i).getSingleValueBlock(i);
+                }
+                else {
+                    nullCount = rowBlock.getChildren().get(i).getLong(i, 0);
+                }
+                nullCounts.put(field.getName().orElseThrow(), toNullCount(field.getType(), nullCount));
```

Comment:
Personally, this would be more readable if the recursive call only happens in the `RowType` block.